src/averageAges.js

```diff
@@ -55,7 +79,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const diferAge = [];
+  let peopleList = people;
+
+  if (onlyWithSon) {
+    peopleList = people.filter(person => person.sex === 'm');
+  }
+
+  peopleList.forEach((person) => {
+    const mother = people.find(
+      somebody => somebody.name === person.mother || false
```

Comment:
what `find` method returns if element isn't found? Do we really need to explicitly set `false` here?